### PR TITLE
[FEDEXSHIP-42] Added more postal code regex validations

### PR DESCRIPTION
**Pull request by Arturo777**
Added more postal code regex validations

### CODE REVIEW

**Review by polishq — CHANGES_REQUESTED** [review]
@Arturo777 Don't forget this other change we discussed on Slack: https://vtex.slack.com/archives/CFJU5HYTV/p1661180823238569?thread_ts=1659547170.177039&cid=CFJU5HYTV

File: dotnet/Data/Constants.cs [hunk 1]
```diff
@@ -17,14 +17,55 @@ public static class Constants
         public const string PACKING_ACCESS_KEY = "AccessKey";
 
         public static readonly ImmutableDictionary<string, string> POSTAL_CODE_REGEX= (new Dictionary<string, string> {
-            {"USA", "\\d{5}$"},
-            {"MEX", "\\d{5}$"},
+            {"ARE", ""},
```
Comment (btalma) [inline dotnet/Data/Constants.cs:20]: Did you intend to match an empty pattern?

File: dotnet/Data/Constants.cs [hunk 2]
```diff
@@ -17,14 +17,55 @@ public static class Constants
...
+            {"ECU", "\\d{6}$"},
+            {"ESP", "\\d{5}$"},
+            {"FIN", "\\d{5}$"},
             {"FRA", "\\d{5}$"},
+            {"GBR", "([A-Za-z][A-Ha-hJ-Yj-y]?[0-9][A-Za-z0-9]? ?[0-9][A-Za-z]{2}|[Gg][Ii][Rr] ?0[Aa]{2})$"},
+            {"GTM", "\\d{5}$"},
+            {"HRV", "\\d{5}$"},
+            {"IDN", "\\d{5}$"},
+            {"IND", "\\d{6}$"},
+            {"IRL", ""},
```
Comment (btalma) [inline dotnet/Data/Constants.cs:44]: Empty pattern?

**Review by polishq — APPROVED** [review]
LGTM, just need Cypress to pass

File: dotnet/Data/MerchantSettingsRepository.cs [hunk 3]
```diff
@@ -64,9 +64,11 @@ public async Task<MerchantSettings> GetMerchantSettings()
                 merchantSettings.ItemModals.Add(new ModalMap("WHITE_GOODS", "NONE", false));
                 merchantSettings.ItemModals.Add(new ModalMap("FIREARMS", "ORM_D", false));
             }
-            if (merchantSettings.SlaSettings.Count != 19)
+            if (merchantSettings.SlaSettings.Count != 21)
```
Comment (btalma) [inline dotnet/Data/MerchantSettingsRepository.cs:67]: This will overwrite all current settings - is that okay?

Comment (polishq) [reply dotnet/Data/MerchantSettingsRepository.cs:67]: Maybe we have one condition if the count is 19 where we simply add the two new ones, and another condition if the count is less than 19 where we add all 21 of them?

**Review by polishq — CHANGES_REQUESTED** [review]

File: dotnet/Data/Constants.cs (hunk 1 above)
Comment (polishq) [reply dotnet/Data/Constants.cs:20]: United Arab Emirates (ARE) apparently does not use postal codes. An empty pattern matches everything, so this "works" 🤷

Comment (polishq) [reply dotnet/Data/Constants.cs:20]: Suggestions welcome, though

**Review by btalma — APPROVED** [review]

**Pull request merged**